### PR TITLE
show more details about partition movements under Executor tab

### PR DESCRIPTION
**Pull request by linehrr**
Screen Shot 2019-07-22 at 1 31 41 PM
Screen Shot 2019-07-22 at 1 31 37 PM
Screen Shot 2019-07-22 at 1 31 31 PM
Screen Shot 2019-07-22 at 4 00 17 PM

added detailed table per partition and also progress bar per each type of movement.

also added bootstrap metrics feature button to the monitor page, so ppl can choose to bootstrap the CC state. 
Screen Shot 2019-07-22 at 5 43 24 PM

### CODE REVIEW

**Comment by linehrr** [comment]
@nareshv please take a look, thank you

File: package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
   "dependencies": {
     "axios": "^0.19.0",
     "bootstrap": "^4.3.1",
+    "bootstrap-vue": "^2.0.0-rc.27",
```
Comment (nareshv) [inline package.json]: I had to remove this dependency in the previous versions as i wanted to keep the number of external package dependencies to really minimum. 
Can you use the native bootstrap provided markup to render the elements like its done in the other places ?

File: src/components/Executor.vue [hunk 2]
```diff
@@ -47,9 +48,14 @@
...
           <div class="card-deck mb-3">
             <div class="card">
-              <div class="card-header">Total Data To Move</div>
+              <div class="card-header">
+                Total Data To Move
+              </div>
               <div class="card-body">
-                <p class="card-text"><h1 class="text-primary">{{ ExecutorState.totalDataToMove | formatUnits }}</h1></p>
+                <div v-if='ExecutorState.totalDataToMove'>
+                  <b-progress height="2rem" :value="ExecutorState.finishedDataMovement / ExecutorState.totalDataToMove * 100" max="100" show-progress animated></b-progress>
```
Comment (nareshv) [inline src/components/Executor.vue]: can you use native boostrap4 css instead ?

**Comment by nareshv** [comment]
Thanks for the Patch, left few comments.
Here are some high level ones to consider

1. features available in newer CC versions should be used conditionally with proper messaging for older versions of CC or else it confuses users.
2. Camelcase the table headings
3. Request you to use native boostrap 4 classes as i want to limit the number of external deps to really low.

**Comment by linehrr** [comment]
@nareshv  thanks for the feedback, will change later.

**Comment by linehrr** [comment]
Screen Shot 2019-08-01 at 9 56 16 AM

@nareshv changed progress bar to bootstrap4. 
also changed table headings to Camel cases. 

re: `features available in newer CC versions should be used conditionally with proper messaging for older versions of CC or else it confuses users.`
what particularly did you mean by that?

**Comment by nareshv** [comment]
Thanks for the changes.

Regarding `features available in newer CC versions should be used conditionally with proper messaging for older versions of CC or else it confuses users`

- If the REST API doesn't support these fields we should not show anything on the UI
- If the REST API supports newer fields,  we have to partially show them based on the values present within

This will ensure that the older CC versions with new CCFE doesn't render additional UI fields

**Comment by nareshv** [comment]
Also @linehrr , why are we showing some values as `NaN` it might be confusing to the user.

**Comment by linehrr** [comment]
> why are we showing some values as NaN

this is actually not my code, it was there from the beginning so I can't tell the reason why it showed NaN.

**Comment by nareshv** [comment]
@linehrr , for the previous comment, see how the current code detects new features and conditionally renders these https://github.com/linkedin/cruise-control-ui/blob/master/src/components/KafkaClusterState.vue#L181

**Comment by linehrr** [comment]
@nareshv I see, but I don't think I've added any feature that only new CC has, can you point out ?

**Comment by nareshv** [comment]
Thats from backward compatibility point of view. I am merging this PR. will update if required

**Pull request merged**